osconfig_tests/test_suites/guestpolicies/guest_policies_test_data.go

```diff
@@ -229,6 +229,130 @@ func addPackageInstallFromNewRepoTest(key string) []*guestPolicyTestSetup {
+		recipeTestSetup = append(recipeTestSetup, buildRecipeStepsTestSetup(name, image, "yum", key))
+	}
+	for name, image := range utils.HeadSUSEImages {
+		recipeTestSetup = append(recipeTestSetup, buildRecipeStepsTestSetup(name, image, "zypper", key))
+	}
+	return recipeTestSetup
+}
+
+func buildRecipeInstallTestSetup(name, image, pkgManager, key string) *guestPolicyTestSetup {
+	assertTimeout := 60 * time.Second
+	testName := recipeInstallFunction
+	recipeName := "testrecipe"
+	machineType := "n1-standard-2"
+	if strings.HasPrefix(image, "windows") {
+		machineType = "n1-standard-4"
+	}
+
+	instanceName := fmt.Sprintf("%s-%s-%s-%s", path.Base(name), testName, key, utils.RandString(3))
+	gp := &osconfigpb.GuestPolicy{
+		Recipes: []*osconfigpb.SoftwareRecipe{
+			osconfigserver.BuildSoftwareRecipe(recipeName, "", nil, nil),
+		},
+		Assignment: &osconfigpb.Assignment{GroupLabels: []*osconfigpb.Assignment_GroupLabel{{Labels: map[string]string{"name": instanceName}}}},
+	}
+	ss := getRecipeInstallStartupScript(name, recipeName, pkgManager)
+	return newGuestPolicyTestSetup(image, instanceName, testName, packageInstalled, machineType, gp, ss, assertTimeout)
+}
+
+func buildRecipeStepsTestSetup(name, image, pkgManager, key string) *guestPolicyTestSetup {
+	assertTimeout := 60 * time.Second
+	testName := recipeStepsFunction
+	recipeName := "testrecipe"
+	machineType := "n1-standard-2"
+
+	instanceName := fmt.Sprintf("%s-%s-%s-%s", path.Base(name), testName, key, utils.RandString(3))
+	gp := &osconfigpb.GuestPolicy{
+		Recipes: []*osconfigpb.SoftwareRecipe{
+			osconfigserver.BuildSoftwareRecipe(recipeName, "",
+				[]*osconfigpb.SoftwareRecipe_Artifact{
+					&osconfigpb.SoftwareRecipe_Artifact{
+						AllowInsecure: true,
+						Id:            "copy-test",
+						Artifact: &osconfigpb.SoftwareRecipe_Artifact_Remote_{
+							Remote: &osconfigpb.SoftwareRecipe_Artifact_Remote{
+								Uri: "https://example.com",
+							},
+						},
+					},
+					&osconfigpb.SoftwareRecipe_Artifact{
+						AllowInsecure: true,
+						Id:            "exec-test",
+						Artifact: &osconfigpb.SoftwareRecipe_Artifact_Gcs_{
+							Gcs: &osconfigpb.SoftwareRecipe_Artifact_Gcs{
+								Bucket: tempBucketName,
+								Object: "exec_test",
+							},
+						},
+					},
+					&osconfigpb.SoftwareRecipe_Artifact{
+						AllowInsecure: true,
+						Id:            "tar-test",
+						Artifact: &osconfigpb.SoftwareRecipe_Artifact_Gcs_{
+							Gcs: &osconfigpb.SoftwareRecipe_Artifact_Gcs{
+								Bucket: tempBucketName,
+								Object: "tar_test.tar.gz",
+							},
+						},
+					},
+					&osconfigpb.SoftwareRecipe_Artifact{
+						AllowInsecure: true,
+						Id:            "zip-test",
+						Artifact: &osconfigpb.SoftwareRecipe_Artifact_Gcs_{
+							Gcs: &osconfigpb.SoftwareRecipe_Artifact_Gcs{
+								Bucket: tempBucketName,
+								Object: "zip_test.zip",
+							},
+						},
+					},
+				},
+				[]*osconfigpb.SoftwareRecipe_Step{
```

Comment:
These steps so far are fairly simple and all of them just put a file somewhere, which makes verification easy